cipherscan

```diff
@@ -1369,8 +1364,34 @@ if [ ! -x $OPENSSLBIN ]; then
     if [ "$OUTPUTFORMAT" == "terminal" ]; then
         echo "custom openssl not executable, falling back to system one from $OPENSSLBIN"
     fi
+else
+    tmp=$(echo "quit" | $OPENSSLBIN 2>&1)
+    if [ $? -ne 0 ]; then
+        OPENSSLBIN=$(which openssl)
+        if [ "$OUTPUTFORMAT" == "terminal" ]; then
+            echo "custom openssl fails to execute properly, falling back to system one from $OPENSSLBIN"
+            verbose "execution failure message: $tmp"
+        fi
+    fi
 fi
 
+# use custom config file to enable GOST ciphers if supported by whatever $OPENSSLBIN we ended up with
+if [[ -e $(dirname $0)/openssl.cnf ]]; then
+    tmp=$(OPENSSL_CONF="$(dirname $0)/openssl.cnf" $OPENSSLBIN s_client -help 2>&1 | grep :error:)
```

Comment:
since we're checking in tests for presence of `-no_tlsext` in help and handling of `-curves` right below, it should be possible to combine them, `parse_openssl_output()` style - I'll fix the lack of `-sigalgs` and `-curves` in upstream openssl, so it should in the end be possible to automatically enable them

I also wonder if combining both checks wouldn't be possible (for executability and supported options) - remember that for bulk scanning we're testing if a given option is supported over and over and over again, that just wastes resources